modules/generic_worker/templates/run-generic-worker.sh.multiuser.erb

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# This is a wrapper script for calling generic worker, such that when the worker
+# exits, this script will reboot the machine. Note, in the worker config
+# (/etc/generic-worker.config) we set numberOfTasksToRun to 1 so that the worker
+# will exit after running a single task. This script is then responsible for
+# rebooting the machine.
+
+# If this file exists when the worker starts up, and its content is equal
+# to numberOfTasksToRun, the worker won't exit and machine won't reboot.
+rm -f tasks-resolved-count.txt
+
+export SHELL=/bin/bash
```

Comment:
You can delete this line, it won't set `SHELL` for tasks. See e.g. [this task](https://tools.taskcluster.net/groups/BH_28jjfRN-5ZA-T1t8JUQ/tasks/BH_28jjfRN-5ZA-T1t8JUQ/runs/0/logs/public%2Flogs%2Flive.log). It runs against production, using this version of `run-generic-worker.sh.multiuser.erb` but SHELL is not set in python process. `SHELL` is a [special bash variable](https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-SHELL), that evaluates to the user's shell if the environment variable `SHELL` is not set. Since the spawned process for the task does not inherit the exported environment variables of the generic-worker process, `SHELL` will not be set, and so a task that executes a bash shell will have the bash variable `SHELL` set, but any processes it runs will not have the `SHELL` environment set.